src/main/java/impl/file/FileEncodingWriterImpl.java

```diff
@@ -0,0 +1,43 @@
+package impl.file;
+
+import api.file.FileEncodingWriter;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+public class FileEncodingWriterImpl implements FileEncodingWriter {
+
+    private final Charset defaultCharset;
+
+    public FileEncodingWriterImpl() {
+        defaultCharset = StandardCharsets.UTF_8;
+    }
+
+    @Override
+    public void write(File file, InputStream data, Charset dataEncoding) {
+        write(file, data, dataEncoding, defaultCharset);
+    }
+
+    @Override
+    public void write(File file, InputStream data, Charset dataEncoding, Charset fileEncoding) {
+        var parent = file.getParentFile();
+        if (parent != null && !parent.exists()) {
+            if(!parent.mkdirs()){
+                // It makes sense to throw here... Hmm...
```

Comment:
Yes, because we couldn't run code below if directories weren't created

Reply:
But we don't have any exceptions in the interface method signature, so we sort of can't throw anything proper

Reply:
You are right, It's better to add IOException to the method signature, but I wanted you to check them explicitly. So, ok let's keep this println